packages/paste-website/src/components/Heading.tsx

```diff
@@ -67,15 +65,47 @@ function getHeadingStyles(headingStyle?: HeadingStyle): {} {
   }
 }
 
-const Heading: React.FC<HeadingProps> = props => (
-  <Text {...props} as={props.as} textColor="colorText" {...getHeadingStyles(props.headingStyle)}>
-    {props.children}
-  </Text>
-);
+export interface HeadingProps {
+  as: asTags;
+  headingStyle?: HeadingStyle;
+}
+
+const Heading: React.FC<HeadingProps> = props => {
+  return (
+    <Text {...props} textColor="colorText" {...getHeadingStyles(props.headingStyle)}>
+      {props.children}
+    </Text>
+  );
+};
 
 Heading.defaultProps = {
   as: 'h2',
   headingStyle: 'headingStyle20',
 };
 
-export {Heading};
+const StyledAnchorHyperlink = styled.a`
+  text-decoration: none;
+  margin-left: ${themeGet('space.space30')};
+  color: ${themeGet('textColors.colorTextWeak')};
+
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
+const AnchoredHeading: React.FC<HeadingProps> = props => {
+  // Only generate slugs for headings where children is 'string'
+  if (typeof props.children === 'string') {
+    const id = slugify(props.children);
+    return (
+      <Text {...props} id={id} textColor="colorText" {...getHeadingStyles(props.headingStyle)}>
```

Comment:
nit: I guess you could have just used `Heading` here instead to potentially save any future double refactoring.

```
<Heading {...props} id={id} headingStyle={props.headingStyle}>
  {props.children}
  <SteyledAnchorHyperlink href={...}>#</StyledAnhorHyperlink>
</Heading>
```